qadence/backends/pyqtorch/convert_ops.py

```diff
@@ -104,87 +102,46 @@ def convert_block(
-        Op = getattr(pyq, name)
-        self.operation = Op(block.qubit_support, n_qubits)
-
-    # primitive blocks do not require any parameter value, hence the
-    # second empty argument added here
-    def forward(self, state: torch.Tensor, _: dict[str, torch.Tensor] = None) -> torch.Tensor:
-        return self.apply(self.matrices(), state)
-
-    def matrices(self, _: dict[str, torch.Tensor] = None) -> torch.Tensor:
-        return self.operation.matrix
-
-    def apply(self, matrices: torch.Tensor, state: torch.Tensor) -> torch.Tensor:
-        return self.operation.apply(matrices, state)
-
-
-class ParametricPyQOperation(Module):
-    def __init__(self, n_qubits: int, block: ParametricBlock, config: Configuration):
-        super().__init__()
-        name = block.name[1:] if block.name.startswith("MC") else block.name
-        Op = getattr(pyq, name)
-        self.operation = Op(block.qubit_support, n_qubits)
-        self.param_names = config.get_param_name(block)
-        num_params = len(self.param_names)
-        if num_params == 1:
-
-            def _fwd(state: torch.Tensor, values: dict[str, torch.Tensor]) -> torch.Tensor:
-                return self.apply(self.matrices(values), state)
-
-        else:
-
-            def _fwd(state: torch.Tensor, values: dict[str, torch.Tensor]) -> torch.Tensor:
-                op_params = {key: values[key] for key in self.param_names}
-                max_batch_size = max(p.size() for p in values.values())
-                new_values = {
-                    k: (v if v.size() == max_batch_size else v.repeat(max_batch_size, 1, 1))
-                    for k, v in op_params.items()
-                }
-                return self.apply(self.matrices(new_values), state)
-
-        if config.use_gradient_checkpointing:
-
-            def _forward(state: torch.Tensor, values: dict[str, torch.Tensor]) -> torch.Tensor:
-                return checkpoint(_fwd, state, values, use_reentrant=False)
-
-        else:
-
-            def _forward(state: torch.Tensor, values: dict[str, torch.Tensor]) -> torch.Tensor:
-                return _fwd(state, values)
-
-        self._forward = _forward
-
-    def matrices(self, values: dict[str, torch.Tensor]) -> torch.Tensor:
-        thetas = torch.vstack([values[name] for name in self.param_names])
-        return self.operation.matrices(thetas)
-
-    def apply(self, matrices: torch.Tensor, state: torch.Tensor) -> torch.Tensor:
-        return self.operation.apply(matrices, state)
-
-    def forward(self, state: torch.Tensor, values: dict[str, torch.Tensor]) -> torch.Tensor:
-        return self._forward(state, values)
 
 
 class PyQMatrixBlock(Module):
     def __init__(self, block: MatrixBlock, n_qubits: int, config: Configuration = None):
         super().__init__()
         self.n_qubits = n_qubits
-        self.qubits = block.qubit_support
+        self.qubits = list(block.qubit_support)
```

Comment:
This is to conform with `PyQTorch` type. So, has the decision been taken to use `list[int]` for qubit supports there ?

Reply:
its back to tuple